component/oauth-auth/src/main/java/io/meeds/oauth/service/impl/OAuthRegistrationServiceImpl.java

```diff
@@ -92,7 +98,8 @@ public OAuthRegistrationServiceImpl(PortalContainer container,
 
   @Override
   public boolean isRegistrationOnFly(OAuthProviderType<? extends AccessTokenContext> oauthProviderType) {
-    return registerOnFly.contains(oauthProviderType.getKey());
```

Comment:
In customer contexts, we need registration on the fly even if registrationType is restricted : 

Functionnal need : 
exo is configured with openid or oauth, but not connected to the AD/ldap. Users are not known in the platform until they log the first time.
In this case, the registration type is restricted AND registerOnTheFly contains OPENID.

With this modification, the user cannot log in and create an account, but is it a valid behaviour

Reply:
ok, this was discussed in a Task as functional behavior need. @srenault-meeds , we may missed a special use case, where it's restricted but Open on some Registration providers. WDYT ?

Reply:
Hi, can we discuss it please @rdenarie? 
Indeed, I need to understand the case to make sure we adapt or not.

Reply:
ok done here 119557e9ab18e7af7e0fad2054b2c44d554c065e
Thanks for both.